src/scripts/sdk/ProtocolAPI/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  IFractalProtocolAPI,
+  IFact,
+  IAsyncIteratorCreator,
+} from "@pluginTypes/index";
+
+import SingletonConnection from "@sdk/InpageProvider/connection";
+import ConnectionTypes from "@models/Connection/types";
+
+class FactIterator implements AsyncIterator<any> {
+  private i: number = 0;
+  private factCount: Promise<number>;
+  private factsAPI: FactsAPI;
+  constructor(factsAPI: FactsAPI) {
+    this.factsAPI = factsAPI;
+    this.factCount = factsAPI.totalFacts();
+  }
+  async next(): Promise<any> {
+    if (this.i >= (await this.factCount)) {
+      return { done: true };
+    }
+
+    const value = await this.factsAPI.get(this.i++);
+    return { value, done: false };
+  }
+}
+
+export class PageView implements IFact {
+  timestampMs: number;
+  url: string;
+  constructor(url: string, timestampMs: number) {
+    this.timestampMs = timestampMs;
+    this.url = url;
+  }
+}
```

Comment:
Little strange to have a type for our facts. Our facts really should be just a JS object (if possible). Could wrap them in an object that gave up space to add metadata.

Reply:
If it's okay I will keep it like this until we add different facts, then I will refactor it having a better overview.

Reply:
SGTM.